pkg/controller/controller.go

```diff
@@ -478,9 +478,34 @@ func (c *M3DBController) handleClusterUpdate(cluster *myspec.M3DBCluster) error
 	// If any of the statefulsets aren't ready, wait until they are as we'll get
 	// another event (ready == bootstrapped)
 	for _, sts := range childrenSets {
-		if sts.Spec.Replicas != nil && *sts.Spec.Replicas != sts.Status.ReadyReplicas {
+		if sts.Spec.Replicas == nil {
 			// TODO(schallert): figure out what to do if replicas is not set
-			c.logger.Info("waiting for statefulset to be ready", zap.String("name", sts.Name), zap.Int32("ready", sts.Status.ReadyReplicas))
+			continue
```

Comment:
Can you just `Warn` here and remove the TODO? I haven't seen this happen in the wild so I don't think we need to figure out how to specifically handle it.

Reply:
Yeah sounds good.